pkg/operator/ceph/object/bucket/predicate.go

```diff
@@ -50,10 +50,12 @@ func predicateController(ctx context.Context, c client.Client) predicate.Funcs {
 		},
 
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			if old, ok := e.ObjectOld.(*v1.ConfigMap); ok {
+
+			switch objOld := e.ObjectOld.(type) {
+			case *v1.ConfigMap:
 				if new, ok := e.ObjectNew.(*v1.ConfigMap); ok {
-					if old.Name == controller.OperatorSettingConfigMapName && new.Name == controller.OperatorSettingConfigMapName {
-						if old.Data[rookOBCWatchOperatorNamespace] != new.Data[rookOBCWatchOperatorNamespace] {
+					if objOld.Name == controller.OperatorSettingConfigMapName && new.Name == controller.OperatorSettingConfigMapName {
+						if objOld.Data[rookOBCWatchOperatorNamespace] != new.Data[rookOBCWatchOperatorNamespace] {
 							logger.Infof("%s changed. reconciling bucket controller", rookOBCWatchOperatorNamespace)
```

Comment:
Let's use happy-path-on-the-left methods here also. Return `false` in the (highly unlikely) case that the new/old object don't cast correctly to the right type. Also log an error in that case, so we can debug any failures.